Fix potential security flaw in creation of sockets/directories

git-core opened the pull request:
See commit 8dc2877f8dbdf4161c394c28e714ee3efd645cf5

Well, I fixed last bit I didn't like in su-binary as promised. There is more stuff to fix:
1. Move Requestor cache dir to /dev
    (requires change in corresponding macro only)
    Thus, su-binary won't touch flash, /dev is tmpfs nowadays
2. Rearrange db code
    su-binary acess db only once, so db open/close can be moved to database_check
3. Drop mktemp so gcc stops complaining about "unsafe" function
    The socket is created once per su invocation. We can use pid in the socket name.
    It's neither "safer" nor "weaker"
If you'll agree to accept mentioned changes I could prepare the patches.

Commit:
Report error from read rather than successful completion of close

Commit:
Remove all occurences of req_uid

Commit:
Remove setgroups in allow()

supplementary groups have been dropped already at this time
(introduced by commit 00f1bb547ef219b690a440920c7bc29be2a1d348)

Commit:
Always free result table if sqlite3_get_table succeeds

Commit:
Clean up sources

ChainsDD commented on su.c:
```diff
@@ -378,8 +378,8 @@ int main(int argc, char *argv[])
-    setegid(st.st_gid);
-    seteuid(st.st_uid);
+    setegid(req_uid);
+    seteuid(req_uid);
```
This bit isn't really necessary, as req_uid == st.st_uid == st.st_gid. I should really just remove req_uid as it's not needed.

git-core replied:
Done. Have to rebase the patchset though

Commit:
Fix potential security flaw in creation of sockets/directories

Previous code created a file object with two steps: the creation itself with
default umask/mode and setting necessary permissions then. This approach is known to
lead to a race condition when malicious process can open an object before
permissions is set.

The patch sets creation mask (mask) to 027, thus denying any access from others. Also,
the patch removes all dead code which is not needed after changes mentioned.

ChainsDD commented:
I'll test this when I get home. If we could discuss some more things over gtalk, I'd really appreciate it, I could use some help with a couple of things.

ChainsDD committed:
bump version

ChainsDD commented:
Works in my testing, will send a couple builds out to my testers to see how it works for them before merging it. Doesn't seem to affect too much, so I don't see there being any issues.
